### PR TITLE
JSDK-2952 TypeScript Definitions: DataTrack

### PR DESCRIPTION
**Pull request by PikaJoyce**
[JSDK-2952](https://issues.corp.twilio.com/browse/JSDK-2952)
This is for TypeScript Definitions for LocalDataTrack as well as RemoteDataTrack. I've added in a tsconfig for compiling the tests since it was throwing errors this time for me. I've also started adding in the JSDocs so that the annotations show up when the package is used. Hopefully it will be more helpful to users.

Please take a look and let me know if I've missed anything.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -124,7 +124,7 @@
     "test:framework:react": "npm-run-all test:framework:react:*",
     "test:framework": "npm-run-all test:framework:angular test:framework:no-framework test:framework:react",
     "test": "npm-run-all test:unit test:integration",
-    "test:ts": "tsc ./tsdef/twilio-video-tests.ts",
+    "tsc:ts": "tsc -p tsconfig.json",
```
Comment (mhuynh5757) [inline package.json:127]: Might want to rename this script, maybe just `build:ts`?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "lib": ["es5", "es6", "es2016", "es2017.object", "dom"],
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "sourceMap": false,
+    "strictNullChecks": true
+  },
+  "files": ["tsdef/twilio-video-tests.ts"]
+}
```
Comment (mhuynh5757) [inline tsconfig.json:15]: Should have a new-line at the EOF.
```suggestion
}

```

**Comment by makarandp0** [comment]
@PikaJoyce  I see some warning in the build with new changes. @  
https://app.circleci.com/pipelines/github/twilio/twilio-video.js/2299/workflows/4eff2c21-fdba-4ec6-84aa-dbe5e26782ed/jobs/25473

Maybe we should just disable `no-unused-vars` at the file level for this file? 

LGTM otherwise.

**Review by makarandp0 — APPROVED** [review]
please fix the build warning, LGTM otherwise

**Pull request merged**